### PR TITLE
Handle executable not found in exec

### PR DESCRIPTION
**Pull request by ghost**
This is needed for #14200 to handle when fsck is not found

### CODE REVIEW

**Comment by swagiaal** [comment]
@k8s-merge-robot test this

**Comment by swagiaal** [comment]
@dchen1107 could you take a look at this ?

File: pkg/util/exec/exec.go [hunk 1]
```diff
@@ -81,13 +84,18 @@ func (cmd *cmdWrapper) SetDir(dir string) {
...
-		ee, ok := err.(*osexec.ExitError)
-		if !ok {
-			return out, err
+		if ee, ok := err.(*osexec.ExitError); ok {
+			// Force a compile fail if exitErrorWrapper can't convert to ExitError.
+			var x ExitError = &exitErrorWrapper{ee}
+			return out, x
+		}
+		if ee, ok := err.(*osexec.Error); ok {
+			error := ee.Err
```
Comment (ashcrow) [inline pkg/util/exec/exec.go]: nit: Why set ```error``` instead of just using ```ee.Err```?

File: pkg/util/exec/exec.go [hunk 2]
```diff
@@ -21,6 +21,9 @@ import (
 	"syscall"
 )
 
+// ErrExecutableNotFound is returned if the executable is not found.
+var ErrExecutableNotFound = osexec.ErrNotFound
```
Comment (ashcrow) [inline pkg/util/exec/exec.go]: At first this bugged me but now I see it does make the code easier to read (```osexec.ErrNotFound``` isn't very descriptive). :+1:

**Comment by ashcrow** [comment]
@dchen1107 this looks good to me. It's ready for an official review.

File: pkg/util/exec/exec.go (hunk 1 above)
Comment (swagiaal) [inline pkg/util/exec/exec.go]: Fixed.

File: pkg/util/exec/exec.go (hunk 2 above)
Comment (swagiaal) [inline pkg/util/exec/exec.go]: > osexec.ErrNotFound isn't very descriptive

Yeah. In addition this follows this package's general notion of abstracting away os/exec. Currently users of this package don't have to import os/exec at all.

**Comment by swagiaal** [comment]
@k8s-bot unit test this

**Comment by wojtek-t** [comment]
@k8s-bot test this please

**Comment by wojtek-t** [comment]
@k8s-bot test this please

**Comment by ashcrow** [comment]
\o/

**Pull request merged**